File: yoga-backend/models/YogaClass.js
const mongoose = require('mongoose');

const yogaClassSchema = new mongoose.Schema({
    dayOfWeek: {
        type: String,
        required: true
    },
    time: {
        type: String,
        required: true
    },
    capacity: {
        type: Number,
        required: true
    },
    duration: {
        type: Number,
        required: true
    },
    price: {
        type: Number,
        required: true
    },
    type: {
        type: String,
        required: true
    },
    description: {
        type: String
    },
    lastSynced: {
        type: Date,
        default: Date.now
    }
});

module.exports = mongoose.model('YogaClass', yogaClassSchema);

File: app/src/main/java/com/example/yogaclassmanagement/YogaClass.java
package com.example.yogaclassmanagement;

import java.io.Serializable;

public class YogaClass {
    private String id;
    private String dayOfWeek;
    private String time;
    private int capacity;
    private int duration;
    private double price;
    private String type;
    private String description;

    // Constructor
    public YogaClass(String dayOfWeek, String time, int capacity,
                     int duration, double price, String type, String description) {
        this.dayOfWeek = dayOfWeek;
        this.time = time;
        this.capacity = capacity;
        this.duration = duration;
        this.price = price;
        this.type = type;
        this.description = description;
    }

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getDayOfWeek() { return dayOfWeek; }
    public String getTime() { return time; }
    public int getCapacity() { return capacity; }
    public int getDuration() { return duration; }
    public double getPrice() { return price; }
    public String getType() { return type; }
    public String getDescription() { return description; }
}
